auacm/app/modules/submission_manager/views.py

```diff
@@ -69,3 +69,44 @@ def submit():
     return serve_response({
         'submissionId': attempt.job
     })
+
+
+@app.route('/api/submit/job/<int:job_id>')
+def get_submit(job_id):
+    """Get the current status of a submission based on its id"""
+    submit = (session.query(models.Submission)
+              .filter(models.Submission.job == job_id).first())
+
+    if not submit:
+        return serve_error('Submission with job id ' + str(job_id) +
+                           ' not found', 401)
+
+    return serve_response(submit.to_dict())
+
+
+@app.route('/api/submit/user/<username>')
+@app.route('/api/submit/user/<username>/<int:limit>')
```

Comment:
I agree these need to be changed to query string parameters. Some comments,

* I'm perfectly content leaving out the call to get the current user's submit. It's a shortcut, and I don't think it's an undue burden to ask the client to keep up with the current user. If you really disagree, I'd lean toward `/api/submit/me` (only because `me` is used elsewhere in the API), though I don't love it.
* I don't like putting submissions under the users route. It's very tempting, but I think submits and users are separate actors in the design. After all, we could have users down the road that don't even associate with APT but are still members on the site (which may mean we need to rethink how I did the ranklist).
* I agree that the `submit` route should be changed, though maybe not in this PR. I'd lean to making it `submission` rather than `submissions` just because there's nothing inherently plural about it.